### PR TITLE
IBP-4489 Location and Method Dropdown Code Clean Up

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js [hunk 1]
```diff
@@ -1,108 +1,7 @@
...
-	}
-
-	if ($('#namingConvention').val() === '3' && $('#breedingMethodId').val() === '0') {
-		showInvalidInputMessage(msgMethodError);
-		return false;
-	}
-	return true;
-}
-
+// TODO: Delete
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js:4]: Can we delete this now?

File: src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js [hunk 2]
```diff
@@ -158,15 +57,7 @@ function showCorrectLocationCombo() { //TODO ADVANCE AND FIELDBOOK-COMMONS
...
-function setFavoriteMethodCheckbox(){//TODO ADVANCE
-	if (methodSuggestionsDerivativeAndMaintenanceFavorite.length > 0) {
-		$('#showFavoriteMethod').prop('checked', true);
-	}
-	else{
-		$('#showFavoriteMethod').prop('checked', false);
-	}
-}
-
+// TODO: Delete
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js:60]: Can we delete this too now?

File: src/main/webapp/WEB-INF/static/js/trialmanager/location/locationService.js [hunk 3]
```diff
@@ -23,6 +23,32 @@
 
 		}]);
 
+	angular.module('manageTrialApp').factory('locationModalService', ['$http', 'studyContext', '$uibModal',
+		function ($http, studyContext, $uibModal) {
+
+			//url.program.breeding.methods=ibpworkbench/content/ProgramMethods?programId=
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/static/js/trialmanager/location/locationService.js:29]: delete this comment?

File: src/main/webapp/WEB-INF/static/js/trialmanager/subobservations/subObservationSet.js [hunk 4]
```diff
@@ -1710,6 +1710,10 @@
 						$scope.observation.onOpenClose(isOpen);
 					}
 
+					$scope.methodChanged = function (item, model) {
+
```
Comment (darla-leafnode) [inline src/main/webapp/WEB-INF/static/js/trialmanager/subobservations/subObservationSet.js:1714]: Is this supposed to be an empty method? If so put a comment why it's empty

**Comment by darla-leafnode** [comment]
Hi @abatac - can you review methods in FieldbookServiceImpl (and search elswhere) that use this logic if it's still relevant:
`if (DataType.LOCATION.equals(variable.getScale().getDataType())) {
				// for location, we get all since it is for saving, so we would
				// be able to set the name properly
				possibleValues = this.getLocations(isGetAllRecords);
			} else {
				possibleValues = this.getAllPossibleValues(variable);
			}`
	If not, I suggest we remove all such codes and the services they call - all the way to FieldbookService in Middleware.

File: src/main/java/com/efficio/fieldbook/web/trial/controller/AdvancingController.java [hunk 5]
```diff
@@ -102,7 +102,9 @@
 
 	private static final String UNIQUE_ID = "uniqueId";
 
-	/** The Constant URL. */
+	/**
+	 * The Constant URL.
+	 */
 	protected static final String URL = "/StudyManager/advance/study";
 
 	private static final String MODAL_URL = "StudyManager/advanceStudyModal";
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/trial/controller/AdvancingController.java:110]: MODAL_URL is unused

File: src/main/java/com/efficio/fieldbook/web/common/controller/LocationsController.java [hunk 6]
```diff
@@ -73,7 +51,7 @@ public Long getCurrentProgramID(final HttpServletRequest request) {
 		final String programUUID = this.contextUtil.getCurrentProgramUUID();
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/common/controller/LocationsController.java:51]: As discussed, please check if you can use the location selector for Add Field module so we can delete this class already

File: src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js (hunk 1 above)
Comment (abatac) [reply src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js]: Done

File: src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js (hunk 2 above)
Comment (abatac) [reply src/main/webapp/WEB-INF/static/js/studyManager/studyManager.js]: Done

File: src/main/webapp/WEB-INF/static/js/trialmanager/location/locationService.js (hunk 3 above)
Comment (abatac) [reply src/main/webapp/WEB-INF/static/js/trialmanager/location/locationService.js]: Done

File: src/main/webapp/WEB-INF/static/js/trialmanager/subobservations/subObservationSet.js (hunk 4 above)
Comment (abatac) [reply src/main/webapp/WEB-INF/static/js/trialmanager/subobservations/subObservationSet.js]: This is done, I just made the onMethodSelect in methods-select component optional, so we don't need to pass an empty function.

File: src/main/java/com/efficio/fieldbook/web/trial/controller/AdvancingController.java (hunk 5 above)
Comment (abatac) [reply src/main/java/com/efficio/fieldbook/web/trial/controller/AdvancingController.java]: Done

File: src/main/java/com/efficio/fieldbook/web/common/controller/LocationsController.java (hunk 6 above)
Comment (abatac) [reply src/main/java/com/efficio/fieldbook/web/common/controller/LocationsController.java]: Done

**Comment by abatac** [comment]
> Hi @abatac - can you review methods in FieldbookServiceImpl (and search elswhere) that use this logic if it's still relevant:
> `if (DataType.LOCATION.equals(variable.getScale().getDataType())) { // for location, we get all since it is for saving, so we would // be able to set the name properly possibleValues = this.getLocations(isGetAllRecords); } else { possibleValues = this.getAllPossibleValues(variable); }`
> If not, I suggest we remove all such codes and the services they call - all the way to FieldbookService in Middleware.

I think it's too risky to change this right now. It will affect the Import Study as well as the Manage Settings. There will be a lot of cleanup related to this. I'll create a separate ticket to deal with this.

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**